ykbh/src/lib.rs

```diff
@@ -58,34 +80,62 @@ impl SIRInterpreter {
             return;
         }
 
-        for stmt in body.blocks[0].stmts.iter() {
-            match stmt {
-                Statement::MkRef(dest, src) => self.mkref(dest, src),
-                Statement::DynOffs { .. } => todo!(),
-                Statement::Store(dest, src) => self.store(dest, src),
-                Statement::BinaryOp { .. } => todo!(),
-                Statement::Nop => {}
-                Statement::Unimplemented(_) | Statement::Debug(_) => todo!(),
-                Statement::Cast(..) => todo!(),
-                Statement::Call(..) | Statement::StorageDead(_) => unreachable!(),
+        loop {
+            for stmt in body.blocks[self.bbidx].stmts.iter() {
```

Comment:
The bbidx won't change in this loop, right? I wonder if rust will be smart enough to lift the `body.blocks[self.bbidx]` outside the loop? Should we do it manually?

Reply:
LLVM should be smart enough to apply LICM **if** optimizations were enabled. Because they aren't, I think it won't get hoisted.

Reply:
Good point!

Reply:
Fixed in 8e04201.